velox/type/Timestamp.cpp

```diff
@@ -71,6 +71,19 @@ void Timestamp::toGMT(const tz::TimeZone& zone) {
   seconds_ = sysSeconds.count();
 }
 
+void Timestamp::toGMT(const tz::TimeZone& zone, int32_t secondsOffset) {
```

Comment:
The idea is that if you want to do a timezone offset conversion, you should use this TimeZone constructor and let the TimeZone object handle the conversion internally:

https://github.com/facebookincubator/velox/blob/main/velox/type/tz/TimeZoneMap.h#L98

Looks like to support this case we should first change the TimeZone class to support std::chrono::second instead of std::chrono::minute, then do the appropriate plumbing. If you change it there, you probably won't need to make any changes to the Timestamp class.

Reply:
@pedroerp, yes, if we truly need to support seconds offset, it's better to change TimeZone class to use seconds for offset time representation. My question is, do we really need to support that kind of timezone? As far as I know, there is no timezone with seconds offset in real world.

Reply:
Hi @pedroerp. Thank you for your suggestion. I also @ you in#11411 to ask about your thoughts on the timezone formats supported by Velox can be the union of Presto and Spark. What do you think? Velox supporting more timezone formats shouldn't affect users' usage, and I can move Spark's normalization code into the `TimeZoneMap`.

Reply:
>My question is, do we really need to support that kind of timezone? As far as I know, there is no timezone with seconds offset in the real world.

@PHILO-HE I asked the same question at some point, but it was pointed out that this is something (somehow) supported by Spark. So if we absolutely need to support it, it would probably better to do so within the TimeZone class, so at least it is a bit more encapsulated.

Reply:
@liujiayi771  let me get a hold on @kevinwilfong who has recently changed that code and has more context how this code is organized now.